pkg/burst/pump.go

```diff
@@ -12,40 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package jobs
+package burst
 
 import (
+	"context"
+	"math/rand/v2"
 	"time"
+)
 
-	"github.com/scylladb/gemini/pkg/stop"
+const ChannelSize = 10000
 
-	"go.uber.org/zap"
-	"golang.org/x/exp/rand"
-)
+func work(ctx context.Context, pump chan<- time.Duration, chance int, sleepDuration time.Duration) {
+	defer close(pump)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			sleep := time.Duration(0)
 
-func NewPump(stopFlag *stop.Flag, logger *zap.Logger) chan time.Duration {
-	pump := make(chan time.Duration, 10000)
-	logger = logger.Named("Pump")
-	go func() {
-		logger.Debug("pump channel opened")
-		defer func() {
-			close(pump)
-			logger.Debug("pump channel closed")
-		}()
-		for !stopFlag.IsHardOrSoft() {
-			pump <- newHeartBeat()
-		}
-	}()
+			if rand.Int()%chance == 0 {
+				sleep = sleepDuration
+			}
 
-	return pump
+			pump <- sleep
```

Comment:
Why it should sleep? Can you explain me exactly how it acts?

Reply:
Burst mechanism, makes a user of this, sleep after `change` is hit. practically if you make `change = 10`.

It will sleep approximatly one in 10. This is just to not overload the server